runtime/src/stakes.rs

```diff
@@ -225,23 +225,44 @@ impl Stakes<StakeAccount> {
     /// full account state for respective stake pubkeys. get_account function
     /// should return the account at the respective slot where stakes where
     /// cached.
-    pub(crate) fn new<F>(stakes: &Stakes<Delegation>, get_account: F) -> Result<Self, Error>
+    pub(crate) fn new<F>(stakes: Stakes<Delegation>, get_account: F) -> Result<Self, Error>
     where
         F: Fn(&Pubkey) -> Option<AccountSharedData>,
     {
-        let stake_delegations = stakes.stake_delegations.iter().map(|(pubkey, delegation)| {
-            let Some(stake_account) = get_account(pubkey) else {
-                return Err(Error::StakeAccountNotFound(*pubkey));
-            };
-            let stake_account = StakeAccount::try_from(stake_account)?;
-            // Sanity check that the delegation is consistent with what is
-            // stored in the account.
-            if stake_account.delegation() == *delegation {
-                Ok((*pubkey, stake_account))
-            } else {
-                Err(Error::InvalidDelegation(*pubkey))
-            }
-        });
+        let mut voter_pubkeys = HashSet::new();
+        let stake_delegations = stakes
+            .stake_delegations
+            .into_iter()
+            .map(|(pubkey, delegation)| {
+                let Some(stake_account) = get_account(&pubkey) else {
+                    return Err(Error::StakeAccountNotFound(pubkey));
+                };
+
+                // Assert that all valid vote-accounts referenced in
+                // stake delegations are already cached.
```

Comment:
I've just moved it around, but I'm not 100% sure what this comment means by "cached". Does it mean it's inside `stakes.vote_accounts`?

Reply:
I understood what this means and updated it